### DISCUSSION TITLE
How to export Lagrangian data

### DISCUSSION THREAD

**Discussion by AlexandreMinotCSI** [CGNS/CGNS#682] [CGNStalk]
Hello,

I have "classical" Eulerian CFD results on an unstructured mesh: One base, one zone, unstructured elements, a flow_solution node containing results for each element. I would also like to export Lagrangian particules, and results for each particules. Each particule is an (x,y,z) position, and multiple values (say temperature, velocity, etc...) for each position. I cannot find the right way to export those particules in CGNS. Does anyone have recommandations?

Many thanks,
Alexandre

**Comment by poinot** [comment]
C2 - Confidential

Hi Alexandre :)

you can use the unstructured NODE element which has its own XYZ coordinates, we already used this element type for the same purpose.
check if it can fit to your own needs.

::Marc

Marc POINOT
Senior Expert Software Development
DST/MADS/WILD

Safran Tech - Paris-Saclay
Rue des Jeunes Bois, Châteaufort
78114 Magny-Les-Hameaux, France
www.safran-group.com<http://www.safran-group.com/>

[SAFRAN]

[ABONNEZ-VOUS]<http://www.safran-group.com/subscription-account>

[Twitter]<http://twitter.com/SAFRAN>

[Facebook]<http://www.facebook.com/GroupeSafran>

[LinkedIn]<http://www.linkedin.com/company/safran>




De : AlexandreMinotCSI ***@***.***>
Envoyé : lundi 2 janvier 2023 16:57
À : CGNS/CGNS ***@***.***>
Cc : Subscribed ***@***.***>
Objet : [CGNS/CGNS] How to exporting Lagrangian data (Discussion #682)


Hello,

I have "classical" Eulerian CFD results on an unstructured mesh: One base, one zone, unstructured elements, a flow_solution node containing results for each element. I would also like to export Lagrangian particules, and results for each particules. Each particule is an (x,y,z) position, and multiple values (say temperature, velocity, etc...) for each position. I cannot find the right way to export those particules in CGNS. Does anyone have recommandations?

Many thanks,
Alexandre

-
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/CGNS/CGNS/discussions/682__;!!Dl6pPzL6!aXtrN-Dr7YVlSQRiJcXnR3nAOalOm7hmjJswfWjJsF54wA4gxEyLLSTYCn2d01maY32kyTotUckFlQZwI2-QEjrBv_2XtA$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/ABQJ3P5HT4TIF7PG2LCPQE3WQL3FBANCNFSM6AAAAAATO5UTYE__;!!Dl6pPzL6!aXtrN-Dr7YVlSQRiJcXnR3nAOalOm7hmjJswfWjJsF54wA4gxEyLLSTYCn2d01maY32kyTotUckFlQZwI2-QEjoDQNPx6g$>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

#
" Ce courriel et les documents qui lui sont joints peuvent contenir des informations confidentielles, être soumis aux règlementations relatives au contrôle des exportations ou ayant un caractère privé. S'ils ne vous sont pas destinés, nous vous signalons qu'il est strictement interdit de les divulguer, de les reproduire ou d'en utiliser de quelque manière que ce soit le contenu. Toute exportation ou réexportation non autorisée est interdite Si ce message vous a été transmis par erreur, merci d'en informer l'expéditeur et de supprimer immédiatement de votre système informatique ce courriel ainsi que tous les documents qui y sont attachés."
******
" This e-mail and any attached documents may contain confidential or proprietary information and may be subject to export control laws and regulations. If you are not the intended recipient, you are notified that any dissemination, copying of this e-mail and any attachments thereto or use of their contents by any means whatsoever is strictly prohibited. Unauthorized export or re-export is prohibited. If you have received this e-mail in error, please advise the sender immediately and delete this e-mail and all attached documents from your computer system."
#

**Comment by MicK7** [comment]
There was also a discussion to make a CPEX for this kind of situation:
https://github.com/CGNS/CGNS/issues/578

**Comment by AlexandreMinotCSI** [comment]
Hi Marc, Hi Mickael,

Thanks you for your answers. Indeed, I had read the CPEX. It looks more advanced than what we need, as it aims to store information to solve for particule interactions as well. We could use a subset of that extension should it be accepted though.

For our current need, we only really need 1D elements and associated FlowSolution. Our initial thoughts were to use NODE elements, as Marc is suggesting. However, our Base is of dimension (3, 3), as it contains 3D elements. Marc, are you suggesting we add NODE elements to our existing zone, despite them being under a base of dimension (3, 3)?

Best,
Alexandre

**Comment by poinot** [comment]
C2 - Confidential

Yes you are right, you should have the same dimensions... unless you try a ZoneSubRegion ?

::Marc

Marc POINOT
Senior Expert Software Development
DST/MADS/WILD

Safran Tech - Paris-Saclay
Rue des Jeunes Bois, Châteaufort
78114 Magny-Les-Hameaux, France
www.safran-group.com<http://www.safran-group.com/>

[SAFRAN]

[ABONNEZ-VOUS]<http://www.safran-group.com/subscription-account>

[Twitter]<http://twitter.com/SAFRAN>

[Facebook]<http://www.facebook.com/GroupeSafran>

[LinkedIn]<http://www.linkedin.com/company/safran>




De : AlexandreMinotCSI ***@***.***>
Envoyé : mardi 3 janvier 2023 16:12
À : CGNS/CGNS ***@***.***>
Cc : POINOT Marc (SAFRAN) ***@***.***>; Comment ***@***.***>
Objet : Re: [CGNS/CGNS] How to export Lagrangian data (Discussion #682)


Hi Marc, Hi Mickael,

Thanks you for your answers. Indeed, I had read the CPEX. It looks more advanced than what we need, as it aims to store information to solve for particule interactions as well. We could use a subset of that extension should it be accepted though.

For our current need, we only really need 1D elements and associated FlowSolution. Our initial thoughts were to use NODE elements, as Marc is suggesting. However, our Base is of dimension (3, 3), as it contains 3D elements. Marc, are you suggesting we add NODE elements to our existing zone, despite them being under a base of dimension (3, 3)?

Best,
Alexandre

-
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/CGNS/CGNS/discussions/682*discussioncomment-4582763__;Iw!!Dl6pPzL6!fkcpgkJzfTluaCYCG19ZYMzTJw7ko6ljxMm4euySztLgmlBdhdvM2wdG16m7fs48vX1kusUaDaIRsME5nmUbiJUfkg3s4Q$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/ABQJ3P7NTALZN4RCJ56WVFLWQQ6UTANCNFSM6AAAAAATO5UTYE__;!!Dl6pPzL6!fkcpgkJzfTluaCYCG19ZYMzTJw7ko6ljxMm4euySztLgmlBdhdvM2wdG16m7fs48vX1kusUaDaIRsME5nmUbiJWgvgyvmQ$>.
You are receiving this because you commented.Message ID: ***@***.******@***.***>>

#
" Ce courriel et les documents qui lui sont joints peuvent contenir des informations confidentielles, être soumis aux règlementations relatives au contrôle des exportations ou ayant un caractère privé. S'ils ne vous sont pas destinés, nous vous signalons qu'il est strictement interdit de les divulguer, de les reproduire ou d'en utiliser de quelque manière que ce soit le contenu. Toute exportation ou réexportation non autorisée est interdite Si ce message vous a été transmis par erreur, merci d'en informer l'expéditeur et de supprimer immédiatement de votre système informatique ce courriel ainsi que tous les documents qui y sont attachés."
******
" This e-mail and any attached documents may contain confidential or proprietary information and may be subject to export control laws and regulations. If you are not the intended recipient, you are notified that any dissemination, copying of this e-mail and any attachments thereto or use of their contents by any means whatsoever is strictly prohibited. Unauthorized export or re-export is prohibited. If you have received this e-mail in error, please advise the sender immediately and delete this e-mail and all attached documents from your computer system."
#

**Comment by AlexandreMinotCSI** [comment]
Hum. It is my understanding that ZoneSubRegions are used to designate a subset of existing elements, not add elements. For example, a few 3D elements among all 3D elements, or a few faces of existing cells. We still need a way to add 1D elements somewhere. Ultimately, particules are an overset to the existing mesh, not a subset. We do not know explicitly which 3D cell the particule is in; that would need to be computed.

**Comment by AlexandreMinotCSI** [comment]
I think ultimately the struggle is the following: Topologically, particules are not linked to the 3D mesh. They are a completely different set of coordinates and associated solutions. So the standard doesn't want to put them together. The standard says: Since particules are not topologically linked to the 3D mesh, you should put them in their own 1D base. However, the physicist in us say: But particules are _in_ my existing Zone_t, and they were influenced by the flow, so it doesn't make sense from a CFD point of view to put them in a different base. This is more a dilemma if your are in the solver and are solving flow-particule interactions (hence the CPEX I think).
